---
title: "MIcrobiome data analysis"
author: "Heng-An"
date: "8/12/2019"
 output:  
      html_document:  
        keep_md: true 
---

```{r}
#Packages needed for microbiome analysis
library("phyloseq")
library("dplyr")
library("ggplot2")
library("grid")
library("permute")
library("lattice")
library("vegan")
library("Biostrings")
library("BiocGenerics")
library("parallel")
library ("DECIPHER")
library("ape")
library("phangorn")
library("base")
library("MASS")
library("Biobase")
library("DESeq")
library("DESeq2")
library ("rlang")
library("ape")
```

```{r}
#set the working directery
setwd("D:/PHD/Soybean microbiome_Nanopore/Microbiome_analysis_in_R/phyloseq_input")

#read in multiple files (Feature tabale.txt and Frequency table.txt) at the sample time
#read in the sample_metadata table
getwd()
setwd("D:/PHD/Soybean microbiome_Nanopore/Microbiome_analysis_in_R/phyloseq_input/featuretable_ID95")
temp = list.files(pattern="*.txt")
temp
for (i in 1:length(temp)) assign(temp[i], read.csv(temp[i], header = TRUE, row.names=1, sep=""))
```


```{r}
#the freqyency talbe and feature table need to transform to matirx
BC76_OTU <- data.matrix(BC76.frequencytable.txt)
BC76_TAX <- as.matrix (BC76.featuretable.txt)

#Formating the sample metadata table
class(Sample_metadataTest.tsv$Rep)
Sample_metadataTest.tsv$Rep <- as.factor(Sample_metadataTest.tsv$Rep)
sampledata <- sample_data(Sample_metadataTest.tsv)
```

```{r}
#transfrom the OTU_table and Tax_table to phyloseq object 
BC76OTU=otu_table(BC76_OTU, taxa_are_rows = TRUE)
BC76TAX=tax_table(BC76_TAX)
physeq76 = phyloseq(BC76OTU, BC76TAX)
```

```{r}
#merge the phyloseq object 
mergephyseq <- merge_phyloseq(physeq02, physeq13, physeq15, physeq40, physeq75, physeq76, sampledata)
mergephyseq
```


```{r}
#merge the OTU with same tax (Species) level 
mergephyseq_mergeOTU <- tax_glom(mergephyseq, taxrank = "Species")
mergephyseq_mergeOTU_ID95 <- tax_glom(mergephyseq_ID95, taxrank = "Species")

##Show number of taxa before/after agglomeration
ntaxa(mergephyseq)
ntaxa(mergephyseq_mergeOTU)
```

```{r}
#trim the singleton
#should be removing any taxa whose row sums comes to less than 2 irrespective of whether this taxa is present in one or two samples.
mergephyseq_mergeOTU_Trim <- prune_taxa(taxa_sums(mergephyseq_mergeOTU) > 1, mergephyseq_mergeOTU) 
mergephyseq_mergeOTU_Trim

#visualize the trim results by creating an otu_table of singletons (and missing i.e. row sums of 0)
sigleton <- otu_table(prune_taxa(taxa_sums(mergephyseq_mergeOTU_Trim) <= 2, mergephyseq_mergeOTU_Trim))
sigleton
```

```{r}
#generate a phylogenic tree and merge to the phyloseq object
random_tree_trim = rtree(ntaxa(mergephyseq_mergeOTU_Trim), rooted=TRUE, tip.label=taxa_names(mergephyseq_mergeOTU_Trim))
mergephyseq_mergeOTU_Trim <- merge_phyloseq(mergephyseq_mergeOTU_Trim,random_tree_trim)
mergephyseq_mergeOTU_Trim
```

```{r}
#plot the distribution and choosing the visulization tech
plot_violin(physeq, x, title = NULL)
```

```{r}
#plot alpha diversity 
#You must use untrimmed, non-normalized count data for meaningful results
plot_richness(mergephyseq_mergeOTU)
plot_richness(mergephyseq_ID95)
plot_richness(mergephyseq_mergeOTU_ID95)
```

```{r}
#normalized the data 
#rarefy_even_depth: Resample an OTU table such that all samples have the same library size
set.seed()
rarefy_mergephyseq_mergeOTU <- rarefy_even_depth(mergephyseq_mergeOTU, sample.size = min(sample_sums(mergephyseq_mergeOTU)),
rngseed = 1234, replace = TRUE, trimOTUs = TRUE, verbose = TRUE)

rarefy_mergephyseq_mergeOTU

rarefy_mergephyseq_mergeOTU_ID95 <- rarefy_even_depth(mergephyseq_mergeOTU_ID95, sample.size = min(sample_sums(mergephyseq_mergeOTU_ID95)),
                                                 rngseed = 112233, replace = TRUE, trimOTUs = TRUE, verbose = TRUE)

rarefy_mergephyseq_mergeOTU_ID95
rarefy_mergephyseq_mergeOTU_ID95 <- merge_phyloseq(rarefy_mergephyseq_mergeOTU_ID95,random_tree_trim_ID95)
```

```{r}
#rarefraction curve:
#The relationship between the number of reads and the number of OTUs can be described using 
#rarefaction curves. This is actually different concept than rarefaction, and it is done for 
#different reasons. Rarefaction is a subsampling technique meant to correct for uneven sample 
#size whereas rarefaction curves are used to estimate whether all the diveristy of the true 
#community was captured.Each line represents a different sample (i.e. column) and shows how 
#many OTUs are found in a random subsets of different numbers of reads. The rarecurve function 
#from the vegan package will do the random sub-sampling and plot the results for an abundance 
#matrix. For example, the code below plots the rarefaction curve for a single sample

rarecurve(t(otu_table(mergephyseq_mergeOTU)), step=50, cex=0.5)
rarecurve(t(otu_table(rarefy_mergephyseq_mergeOTU)), step=50, cex=0.5)

rarecurve(t(otu_table(mergephyseq_mergeOTU_ID90)), step=50, cex=0.5)
rarecurve(t(otu_table(rarefy_mergephyseq_mergeOTU_ID90)), step=50, cex=0.5)
```




```{r}
#taxa bar plot 
plot_bar(rarefy_mergephyseq_mergeOTU, fill="Phylum")
gp.ch = subset_taxa(rarefy_mergephyseq_mergeOTU, Family == "Mycosphaerellaceae")
plot_bar(gp.ch)
plot_bar(gp.ch, fill="Species")

#the function that remove the black grid
plot_bar2 <- function (physeq, x = "Sample", y = "Abundance", fill = NULL, 
          title = NULL, facet_grid = NULL) 
{
  mdf = psmelt(physeq)
  p = ggplot(mdf, aes_string(x = x, y = y, fill = fill))
  p = p + geom_bar(stat = "identity", position = "stack")
  p = p + theme(axis.text.x = element_text(angle = -90, hjust = 0))
  if (!is.null(facet_grid)) {
    p <- p + facet_grid(facet_grid)
  }
  if (!is.null(title)) {
    p <- p + ggtitle(title)
  }
  return(p)
}

plot_bar2(mergephyseq, fill="Species")
```

```{r}
#plot_heatmap
plot_heatmap(mergephyseq_mergeOTU_Trim, sample.label="Sample")
gpac <- subset_taxa(GlobalPatterns, Phylum=="Crenarchaeota")
```

```{r}
#Test whether the observed number of OTUs differs significantly between Treatment. We make a non-parametric test, the Wilcoxon rank-sum test (Mann-Whitney):
#This need singelton info to run richness analysis
rich = estimate_richness(rarefy_mergephyseq_mergeOTU) 
sample_data(rarefy_mergephyseq_mergeOTU)
rich
pairwise.wilcox.test(rich$Shannon, sample_data(rarefy_mergephyseq_mergeOTU)$Treatment)
pairwise.wilcox.test(rich$Observed, sample_data(rarefy_mergephyseq_mergeOTU)$Treatment)
```

```{r}
# PCoA plot using the unweighted UniFrac as distance 
#beta diversity #need to use normalized data
wunifrac_dist = phyloseq::distance(mergephyseq_mergeOTU_Trim, method="unifrac", weighted=F)
wunifrac_dist

ordination = ordinate(mergephyseq_mergeOTU_Trim, method="PCoA", distance=wunifrac_dist)

plot_ordination(mergephyseq_mergeOTU_Trim, ordination, color="Treatment") + theme(aspect.ratio=1) + geom_point(size = 3)
```


```{r}
#Test whether the treatment differ significantly from each other using the permutational ANOVA (PERMANOVA) analysis:
adonis(wunifrac_dist ~ sample_data(mergephyseq_mergeOTU_Trim)$Treatment)
adonis(wunifrac_dist ~ sample_data(rarefy_mergephyseq_mergeOTU)$Treatment)
adonis(wunifrac_dist ~ sample_data(rarefy_mergephyseq_mergeOTU_ID95)$Treatment)
```

```{r}
# function to find the most abundant taxa
# Goes through a phyloseq object, picks out the most abundant taxa and gives the abundance for each
# and identifies which taxa is most abundant for which sample
# 
find.top.taxa <- function(x,taxa){
  require(phyloseq)
  top.taxa <- tax_glom(x, taxa)
  otu <- otu_table(top.taxa) # remove the transformation if using a merge_sample object
  tax <- tax_table(top.taxa)
  j<-apply(otu,1,which.max)
  k <- j[!duplicated(j)]
  l <- data.frame(tax@.Data[k,]) # note the modification here and the line below
  m <- data.frame(otu@.Data[,k])
  s <- as.name(taxa)
  colnames(m) = l[,taxa]
  n <- colnames(m)[apply(m,1,which.max)]
  m[,taxa] <- n
  return(m)
}

find.top.taxa(top.pdy,"Phylum")

# find more than top abundance taxa
find.top.taxa2 <- function(x,taxa,num){
  require(phyloseq)
  require(magrittr)
  
  top.taxa <- tax_glom(x, taxa)
  otu <- otu_table(top.taxa) # remove the transformation if using a merge_sample object
  tax <- tax_table(top.taxa)
  j1 <- apply(otu,1,sort,index.return=T, decreasing=T) # modifying which.max to return a list of sorted index
  j2 <- lapply(j1,'[[',"ix") # select for index
  
  #k <- j[!duplicated(j)] # Replaced with unique() below
  l <- data.frame(unique(tax@.Data[unlist(j2),]))
  m <- data.frame(otu@.Data[,unique(unlist(j2))])
  #s <- as.name(taxa) # This isn't necessary!
  colnames(m) = l[,taxa]
  n <- apply(m,1,sort,index.return=T, decreasing=T) %>%
    lapply('[[',"ix") %>%  # Extract index
    lapply(head,n=num) # This to returns the top x tax
  # I want to apply a list of list of index to obtains a list of list of translated names
  
  
  # https://stackoverflow.com/questions/31561238/lapply-function-loops-on-list-of-lists-r
  p <- list()
  for(i in 1:length(n)){
    p[[i]]<- colnames(m)[n[[i]]]
  }
  m$taxa <- p # replacing [,taxa], but now the new column is called "taxa" instead of the inputted taxonomic rank
  return(m)
}

# Test
find.top.taxa2(ps,"Genus",25)$taxa # Here are the top 25 most abundant genera per sample
```


